src/client/pages/OfferNew/Introduction/Sidebar/StartDateLabelSwitcher.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import styled from '@emotion/styled'
+import { colorsV3 } from '@hedviginsurance/brand'
+import { externalInsuranceProviders } from '@hedviginsurance/embark'
+import {
+  useExternalInsuranceDataQuery,
+  InsuranceDataCollection,
+} from 'data/graphql'
+import { OfferQuote } from 'pages/OfferNew/types'
+import { useTextKeys } from 'utils/textKeys'
+
+type Props = {
+  dataCollectionId: OfferQuote['dataCollectionId']
+}
+
+const DataCollectedStartDateWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  align-items: baseline;
+  justify-content: space-between;
+  flex-wrap: wrap;
+  text-align: right;
+  padding-right: 1rem;
+`
+const DataCollectedStartDateDescription = styled.span`
+  font-size: 0.75rem;
+  line-height: 0.75rem;
+  color: ${colorsV3.gray700};
+`
+const DataCollectedStartDateValue = styled.span`
+  font-size: 1rem;
+  padding-right: 0.5rem;
+`
+const GenericLabel = styled.span`
+  font-size: 0.875rem;
+  line-height: 1;
+  color: ${colorsV3.gray500};
+`
+const Empty = styled.div`
+  width: 100%;
+`
+
+const getValidCurrentInsurerName = (
+  insuranceData?: InsuranceDataCollection,
+): string | null => {
+  if (insuranceData) {
+    const validExternalInsuranceProviders = externalInsuranceProviders.filter(
+      (provider) => provider.externalCollectionId,
+    )
+
+    const currentInsuranceProvider = validExternalInsuranceProviders.find(
+      ({ externalCollectionId }) => {
+        return (
+          externalCollectionId ===
+          insuranceData.insuranceProvider?.toUpperCase()
+        )
+      },
+    )
+
+    return currentInsuranceProvider?.name || null
+  }
+
+  return null
+}
+
+export const StartDateLabelSwitcher: React.FC<Props> = ({
+  dataCollectionId,
+}) => {
+  const {
+    data: externalInsuranceData,
+    loading,
+  } = useExternalInsuranceDataQuery({
+    variables: {
+      reference: dataCollectionId || '',
+    },
+    skip: !dataCollectionId,
+  })
+  const textKeys = useTextKeys()
+
+  // TODO: fix this when implementing Insurely in Norway 👇
+  const firstCurrentInsurance =
+    externalInsuranceData?.externalInsuranceProvider?.dataCollection[0]
+
+  const renewalDate = firstCurrentInsurance?.renewalDate
+  const currentInsurerName = getValidCurrentInsurerName(firstCurrentInsurance)
+
+  return (
+    <>
+      {loading && <Empty />}
+      {renewalDate && !loading && (
+        <DataCollectedStartDateWrapper>
+          <DataCollectedStartDateValue>
+            {renewalDate}
+          </DataCollectedStartDateValue>
+          <DataCollectedStartDateDescription>
+            {textKeys.START_DATE_EXTERNAL_PROVIDER_SWITCH({
+              insuranceProvider: currentInsurerName,
+            })}
+          </DataCollectedStartDateDescription>
+        </DataCollectedStartDateWrapper>
+      )}
+      {!renewalDate && !loading && (
+        <GenericLabel>
+          {textKeys.SIDEBAR_STARTDATE_CELL_VALUE_SWITCHER()}
+        </GenericLabel>
+      )}
+    </>
+  )
```

Comment:
I'm not sure what I think about this. I kind of like having just one return statement in a component, then when looking at the file you can easily find what will be rendered. Though I can see a point with the JSX being less cluttered. But it's also nice in a way to see what are the exact conditions for these elements to be rendered. 
That's why I'm also not a huge fan of the `? :` syntax - it can get hard to understand what conditionals actually apply for specific elements. 🤔

Reply:
But with all that said - this can definitely be written in a slightly more elegant way 😄

Reply:
Good points! Yes I think that ternaries can be a bit bad for readability for sure and also nesting conditional rendering or multiple conditional statements. In some places I agree it can be a bit hard to find if you do returns in different places if you have a messy component. But I do like the pattern of checking early and for one thing at the time, if you keep it together at the end of the component and it's clear what's being rendered like this for example 😃  :

```
  if (error) {
    return <ErrorMessage />
  }

  if (loading) {
    return <Empty />
  }

  return (
    ....
```

Reply:
Hm I see your point but it's a bit of a Slippery Slope! ☝️ Then suddenly people are returning 50 lines of JSX in the middle of a component!!!
But if we agree on that all return statements are at the bottom of the component (like in your example) and you get severely punished if you put them somewhere else then I guess it can be fine as a principle.

Reply:
I hear you on that! 🙃  As long as we review the code we can prevent that 🎉  Or have a punishment 😅